From 83d61e327df1e2398d5a7fe997107bd660a6f34e Mon Sep 17 00:00:00 2001
From: purofle <yuguo.or@isrc.iscas.ac.cn>
Date: Wed, 8 Apr 2026 22:35:05 +0800
Subject: [PATCH] core-net: Fix ISO C23 errors with strchr()

Since glibc-2.43:

For ISO C23, the functions bsearch, memchr, strchr, strpbrk, strrchr,
strstr, wcschr, wcspbrk, wcsrchr, wcsstr and wmemchr that return pointers
into their input arrays now have definitions as macros that return a
pointer to a const-qualified type when the input argument is a pointer
to a const-qualified type.

Signed-off-by: purofle <yuguo.or@isrc.iscas.ac.cn>
(cherry picked from commit 03afafa6b38f538231ffe0b967baea6ee2daef30)
---
 lib/core-net/socks5-client.c   | 17 +++++++++--------
 lib/roles/http/server/server.c |  3 ++-
 2 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/lib/core-net/socks5-client.c b/lib/core-net/socks5-client.c
index 4d7572a12..308bd64bc 100644
--- a/lib/core-net/socks5-client.c
+++ b/lib/core-net/socks5-client.c
@@ -29,7 +29,8 @@
 int
 lws_set_socks(struct lws_vhost *vhost, const char *socks)
 {
-	char *p_at, *p_colon;
+	const char *p_at, *p_colon_in;
+	char *p_colon;
 	char user[96];
 	char password[96];
 
@@ -47,23 +48,23 @@ lws_set_socks(struct lws_vhost *vhost, const char *socks)
 			goto bail;
 		}
 
-		p_colon = strchr(socks, ':');
-		if (p_colon) {
-			if (lws_ptr_diff_size_t(p_colon, socks) >
+		p_colon_in = strchr(socks, ':');
+		if (p_colon_in) {
+			if (lws_ptr_diff_size_t(p_colon_in, socks) >
 							     sizeof(user) - 1) {
 				lwsl_vhost_err(vhost, "user too long");
 				goto bail;
 			}
-			if (lws_ptr_diff_size_t(p_at, p_colon) >
+			if (lws_ptr_diff_size_t(p_at, p_colon_in) >
 						         sizeof(password) - 1) {
 				lwsl_vhost_err(vhost, "pw too long");
 				goto bail;
 			}
 
 			lws_strncpy(vhost->socks_user, socks,
-				    lws_ptr_diff_size_t(p_colon, socks) + 1);
-			lws_strncpy(vhost->socks_password, p_colon + 1,
-				lws_ptr_diff_size_t(p_at, (p_colon + 1)) + 1);
+				    lws_ptr_diff_size_t(p_colon_in, socks) + 1);
+			lws_strncpy(vhost->socks_password, p_colon_in + 1,
+				lws_ptr_diff_size_t(p_at, (p_colon_in + 1)) + 1);
 		}
 
 		lwsl_vhost_info(vhost, " Socks auth, user: %s, password: %s",
diff --git a/lib/roles/http/server/server.c b/lib/roles/http/server/server.c
index 98f30c702..7c48a0516 100644
--- a/lib/roles/http/server/server.c
+++ b/lib/roles/http/server/server.c
@@ -1256,7 +1256,8 @@ int
 lws_http_proxy_start(struct lws *wsi, const struct lws_http_mount *hit,
 		     char *uri_ptr, char ws)
 {
-	char ads[96], host[96], *pcolon, *pslash, unix_skt = 0;
+	const char *pcolon, *pslash;
+	char ads[96], host[96], unix_skt = 0;
 	struct lws_client_connect_info i;
 	struct lws *cwsi;
 	int n, na;
-- 
2.34.1

